dumpling/tests/s3/run.sh

```diff
@@ -80,21 +80,58 @@ echo "1st records count is ${cnt}"
 # run dumpling with compress option
 export DUMPLING_OUTPUT_DIR=s3://mybucket/dump-compress
 
-run_dumpling --s3.endpoint="http://$S3_ENDPOINT/" --compress "gzip"
+for compressType in "gzip" "snappy" "zst"
+do
+  echo "start ${compressType} test"
+  run_dumpling --s3.endpoint="http://$S3_ENDPOINT/" --compress $compressType
 
-mkdir -p "${HOST_DIR}/compress"
+  mkdir -p "${HOST_DIR}/compress"
 
-bin/mc cp minio/mybucket/dump-compress/s3-schema-create.sql.gz "${HOST_DIR}/compress/s3-schema-create.sql.gz"
-bin/mc cp minio/mybucket/dump-compress/s3.t-schema.sql.gz "${HOST_DIR}/compress/s3.t-schema.sql.gz"
-bin/mc cp minio/mybucket/dump-compress/s3.t.000000000.sql.gz "${HOST_DIR}/compress/s3.t.000000000.sql.gz"
-
-gzip "${HOST_DIR}/compress/s3-schema-create.sql.gz" -d
-diff "${HOST_DIR}/local/s3-schema-create.sql" "${HOST_DIR}/compress/s3-schema-create.sql"
-
-gzip "${HOST_DIR}/compress/s3.t-schema.sql.gz" -d
-diff "${HOST_DIR}/local/s3.t-schema.sql" "${HOST_DIR}/compress/s3.t-schema.sql"
-
-gzip "${HOST_DIR}/compress/s3.t.000000000.sql.gz" -d
-diff "${HOST_DIR}/local/s3.t.000000000.sql" "${HOST_DIR}/compress/s3.t.000000000.sql"
+  if [ "$compressType" = "gzip" ]; then
+    compressFormat="gz"
+  else
+    compressFormat=$compressType
+  fi
+  bin/mc cp minio/mybucket/dump-compress/s3-schema-create.sql.${compressFormat} "${HOST_DIR}/compress/s3-schema-create.sql.${compressFormat}"
+  bin/mc cp minio/mybucket/dump-compress/s3.t-schema.sql.${compressFormat} "${HOST_DIR}/compress/s3.t-schema.sql.${compressFormat}"
+  bin/mc cp minio/mybucket/dump-compress/s3.t.000000000.sql.${compressFormat} "${HOST_DIR}/compress/s3.t.000000000.sql.${compressFormat}"
+
+  case $compressType in
+  "gzip")
+    gzip "${HOST_DIR}/compress/s3-schema-create.sql.gz" -d
```

Comment:
seems we can assign gzip, snappy, zst to `$binary`, and also assign $suffix and use

```bash
    ${binary} "${HOST_DIR}/compress/s3-schema-create.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3-schema-create.sql" "${HOST_DIR}/compress/s3-schema-create.sql"

    ${binary} "${HOST_DIR}/compress/s3.t-schema.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3.t-schema.sql" "${HOST_DIR}/compress/s3.t-schema.sql"

    ${binary} "${HOST_DIR}/compress/s3.t.000000000.sql.${suffix}" -d
    diff "${HOST_DIR}/local/s3.t.000000000.sql" "${HOST_DIR}/compress/s3.t.000000000.sql"
```

Reply:
modified